logos-codegen/src/graph/regex.rs

```diff
@@ -175,10 +174,9 @@ fn is_one_ascii(class: &ClassUnicode) -> bool {
     }
 
     let range = &class.ranges()[0];
-    let start = range.start() as u32;
     let end = range.end() as u32;
 
-    start < 128 && (end < 128 || end == 0x0010_FFFF)
+    end < 128 || end == 0x0010_FFFF
```

Comment:
`range.start() <= range.end()` should always be true, so checking that only `end < 128` should be sufficient. However, I don't know if `end` can be `0x0010_FFFF` and `start >= 128`. If such case exists, then I should go revert that changes.

Any idea @maciejhirsz?

Reply:
> I don't know if `end` can be `0x0010_FFFF` and `start >= 128`

It can if you do exclusion range with non-ascii characters.

Reply:
Thanks !